crates/red_knot_project/src/walk.rs

```diff
@@ -129,7 +129,16 @@ impl<'a> ProjectFilesWalker<'a> {
     {
         let mut paths = paths.into_iter();
 
-        let mut walker = db.system().walk_directory(paths.next()?.as_ref());
+        let mut walker = db
+            .system()
+            .walk_directory(paths.next()?.as_ref())
+            .standard_filters(
+                db.project()
+                    .metadata(db)
+                    .options()
+                    .respect_ignore_files
+                    .unwrap_or(true),
+            );
 
         for path in paths {
```

Comment:
We can do this in another PR but I noticed that Ruff doesn't ignore hidden files. We should do the same.

Reply:
@MichaReiser makes sense - added a new issue for that here https://github.com/astral-sh/ruff/issues/17626